internal/tasks/device_render.go

```diff
@@ -292,6 +292,47 @@ func (t *DeviceRenderLogic) renderConfigItem(ctx context.Context, configItem *ap
 	}
 }
 
+// denyUnsafeDevicePath returns an error if the given absolute device path targets
+// agent-managed or read-only directories/files that must not be written by config providers.
+// Denied paths:
+//   - /var/lib/flightctl (and all subpaths)
+//   - /usr/lib/flightctl (and all subpaths)
+//   - /etc/flightctl/certs (and all subpaths)
+//   - /etc/flightctl/config.yaml (file itself)
+//
+// The check denies the exact file and the roots listed above and any subpath under those roots.
+func denyUnsafeDevicePath(p string) error {
```

Comment:
nit: unsafe is scary
```suggestion
func denyForbiddenDevicePath(p string) error {
```